checks.d/mapreduce.py

```diff
@@ -0,0 +1,545 @@
+                            #     MAPREDUCE_JOBS_PATH,
+                            #     job_id)
+
+                            tags = ['cluster_id:' + str(cluster_id),
+                                    'app_name:' + str(app_name),
+                                    'user_name:' + str(user_name),
+                                    'job_name:' + str(job_name)]
+
+                            self._set_metrics_from_json(tags, job_json, MAPREDUCE_JOB_METRICS)
+
+        return running_jobs
+
+    # def _mapreduce_job_counters_metrics(self, running_jobs, cluster_id):
+    #     '''
+    #     Get metrics for each MapReduce job counter
+    #     '''
+    #     for job_id, tracking_url in running_jobs.iteritems():
+    #         metrics_json = self._rest_request_to_json(tracking_url,
+    #             'counters',
+    #             MAPREDUCE_SERVICE_CHECK)
+
+    #         counter_groups = metrics_json.get('jobCounters', {}).get('counterGroup', [])
+
+    #         for counter_group in counter_groups:
+    #             group_name = counter_group.get('counterGroupName')
+
+    #             counters = counter_group.get('counter', [])
+
+    #             for counter in counters:
+    #                 counter_name = counter.get('name')
+
+    #                 if counter_name is not None:
+
+    #                     tags = ['cluster_id:' + str(cluster_id),
+    #                             'job_id:' + str(job_id),
+    #                             'group_name:' + str(group_name),
+    #                             'counter_name:' + str(counter_name)]
+
+    #                     self._set_metrics_from_json(tags,
+    #                         counter,
+    #                         MAPREDUCE_JOB_COUNTER_METRICS)
+
+    def _mapreduce_task_metrics(self, running_jobs, cluster_id):
+        '''
+        Get metrics for each MapReduce task
+        Return a dictionary of {task_id: 'tracking_url'} for each MapReduce task
+        '''
+
+        for job_id, job_stats in running_jobs.iteritems():
+            metrics_json = self._rest_request_to_json(job_stats['tracking_url'],
+                    'tasks',
+                    MAPREDUCE_SERVICE_CHECK)
+
+            if metrics_json['tasks'] is not None:
+                if metrics_json['tasks']['task'] is not None:
+
+                    for task in metrics_json['tasks']['task']:
+                        task_type = task['type']
+                        task_progress = task['progress']
+
+                        tags = ['cluster_id:' + str(cluster_id),
+                                'app_name:' + job_stats['app_name'],
+                                'user_name:' + job_stats['user_name'],
+                                'job_name:' + job_stats['job_name']
+                                ]
+
+                        if task_type == 'MAP':
+                            self._set_metric('mapreduce.job.task.map.progress', HISTOGRAM, task_progress, tags=tags)
```

Comment:
Just a quick comment: don't forget to add a tag with the `task_type` here if you're sending this metric for the two different task types ;)